test/util/util.go

```diff
@@ -613,52 +613,38 @@ func Expect1PodOfEach(ctx context.Context, client *kubernetes.Clientset, ns stri
 }
 
 func ScaleDeployment(ctx context.Context, client *kubernetes.Clientset, ns string, name string, target int32) error {
+	ginkgo.GinkgoHelper()
 	ginkgo.By(fmt.Sprintf("Scale Deployment %q in namespace %q to %d", name, ns, target))
 	gotSc, err := client.AppsV1().Deployments(ns).GetScale(ctx, name, metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
 	gotSc.Spec.Replicas = target
 	_, err = client.AppsV1().Deployments(ns).UpdateScale(ctx, name, gotSc, metav1.UpdateOptions{})
-	if err != nil {
-		return err
-	}
-	gomega.EventuallyWithOffset(1, func() error {
+	return err
+}
+
+func WaitForDepolymentAvailability(ctx context.Context, client kubernetes.Interface, ns string, name string) {
+	ginkgo.GinkgoHelper()
+	var target int32
+	gomega.Eventually(func() error {
 		gotDeploy, err := client.AppsV1().Deployments(ns).Get(ctx, name, metav1.GetOptions{})
 		if err != nil {
 			return err
 		}
-		labelSelector := metav1.FormatLabelSelector(gotDeploy.Spec.Selector)
-		podList, err := client.CoreV1().Pods(ns).List(ctx, metav1.ListOptions{
-			LabelSelector: labelSelector,
-		})
-		if err != nil {
-			return err
-		}
-		if len(podList.Items) != int(target) {
-			return fmt.Errorf("pod count not converged yet")
-		}
```

Comment:
I hope removing these will not make the test flaky by allowing coexistence of old and new pods, as described [here](https://github.com/kubestellar/kubestellar/pull/2356#discussion_r1697294754). We can merge and see.

Reply:
Co-existence for some time is deliberately intended functionality of `Deployment` objects. But the point of the utility code is to wait until the transition is complete.